andy_api/api/intent_processing/move_piece_from.py

```diff
@@ -39,7 +41,17 @@ def handle(session_id, intent_model, board_str):
         from_location = intent_model.parameters["fromLocation"]
         static_choice = get_random_choice(HAPPY_PATH_RESPONSES)
 
-        # TODO: add check that the piece exists
+        board = chess.Board(board_str)
+
+        # check if space contains piece
+        if(board.piece_at(from_location) == None):
+            return "What are you trying to move- there's nothing there?", False, updated_board_str
+
+        # check that player owns piece
+        player_color = board.color_at(from_location)
+        from_color = board.turn()
+        if(player_color != from_color):
+            return "Don't touch my piece!", False, updated_board_str
```

Comment:
Create a dedicated list of responses (like `HAPPY_PATH_RESPONSES`) and get a random response from there, rather than having just a static string.